packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -99,80 +100,106 @@ class SpellCheckConfiguration {
+      offset = adjustedSpanStart - currentSpan.range.start;
       correctedSpellCheckResults.add(adjustedSpan);
     } else {
-      // Search for currentSpan in new text and if found, apply it to new text
-      // by adding it to the list of corrected results.
-      regex = RegExp('\\b$currentSpanText\\b');
-      foundIndex = newText.substring(searchStart).indexOf(regex);
-
-      if (foundIndex >= 0) {
-        foundIndex += searchStart;
-        spanLength = currentSpan.range.end - currentSpan.range.start;
-        searchStart = foundIndex + spanLength;
-        adjustedSpan = SuggestionSpan(
-            TextRange(start: foundIndex, end: searchStart),
-            currentSpan.suggestions
-        );
-        offset = foundIndex - currentSpan.range.start;
+      // Check if word was modified by extension by searching for word
+      // anywhere in the remaing substring of newText left to search.
+      final RegExp modifiedRegex = RegExp(currentSpanText);
+      final int modifiedFoundIndex = newText.substring(searchStart).indexOf(modifiedRegex);
+
+      if (modifiedFoundIndex >= 0) {
+        // Word was modified by extension, so check for next space following the
+        // modified word to find its end index.
+        final RegExp regex = RegExp(' ');
+        final int adjustedSearchStartForSpace = modifiedFoundIndex + searchStart + spanLength;
+        final int foundEndIndex = newText.substring(adjustedSearchStartForSpace).indexOf(regex);
+
+        if (foundEndIndex >= 0) {
+          // Word was found within newText.
+          final int adjustedSpanStart = modifiedFoundIndex + searchStart;
+          final int adjustedSpanEnd = adjustedSearchStartForSpace + foundEndIndex;
+          final SuggestionSpan adjustedSpan = SuggestionSpan(
+            TextRange(start: adjustedSpanStart, end: adjustedSpanEnd),
+            currentSpan.suggestions,
+          );
+
+          // Start search for the next misspelled word at the end of the
+          // adjusted currentSpan.
+          searchStart = adjustedSpanEnd + 1;
+          correctedSpellCheckResults.add(adjustedSpan);
+          // Adjust offset to reflect the difference between where currentSpan
+          // was positioned in resultsText versus where the modified version
+          // is positioned in newText.
+          offset = (adjustedSpanEnd - adjustedSpanStart) - (currentSpan.range.end - currentSpan.range.start) + (adjustedSpanStart - currentSpan.range.start);
+        } else {
+          // Word was found at the end of newText.
+          final int adjustedSpanStart = modifiedFoundIndex + searchStart;
+          final SuggestionSpan adjustedSpan = SuggestionSpan(
+            TextRange(start: adjustedSpanStart, end: newText.length),
+            currentSpan.suggestions,
+          );
 
-        correctedSpellCheckResults.add(adjustedSpan);
+          correctedSpellCheckResults.add(adjustedSpan);
+          break;
+        }
```

Comment:
Any thoughts on the performance of this algorithm?  It seems like it repeatedly uses `substring` in order to reduce the search space (I think substring would be O(1) and therefore a valid optimization) and then `indexOf` to search, which I would think is O(reduced search space) in the worst case.  Probably reasonable given that the misspelled words could have moved anywhere in the string, but let me know if I'm missing anything.

Reply:
I do believe that's right and I had the same concern. I think some optimizations could be made if we make further assumptions about when this method will be needed, e.g. if we assume only one modification was made, then we'd only need to do one search and then use lookups to verify the rest of the results based on the first (and only, in this case) calculated `offset` and `searchStart`. I think it is worth considering, my only concern here is time.

Reply:
Sounds good, it should be fine as-is for now.  We haven't done a lot of work on optimizing for large amounts of text in general in Flutter yet.

Also we probably couldn't ever assume that there was only one modification, even though that's probably the case 99% of the time (because the user pressed a single key on their keyboard).  It's possible that the IME has sent us any wild modification or even an entirely new string.  We encountered the same problem in @Renzo-Olivares's deltas work.